src/bgen/TypeCache.cs

```diff
@@ -0,0 +1,310 @@
+	public Type CGGradient { get; }
+	public Type CGImage { get; }
+	public Type CGImageSource { get; }
+	public Type CGLayer { get; }
+	public Type CGPath { get; }
+	public Type CGVector { get; }
+	public Type DispatchQueue { get; }
+	public Type DispatchData { get; }
+	public Type NSNumber { get; }
+	public Type NSRange { get; }
+	public Type NSString { get; }
+	public Type NSValue { get; }
+	public Type NSZone { get; }
+	public Type SCNMatrix4 { get; }
+	public Type SCNVector3 { get; }
+	public Type SCNVector4 { get; }
+	public Type SecAccessControl { get; }
+	public Type SecIdentity { get; }
+	public Type SecTrust { get; }
+	public Type SecProtocolMetadata { get; }
+	public Type SecProtocolOptions { get; }
+	public Type SecTrust2 { get; }
+	public Type SecIdentity2 { get; }
+	public Type CoreGraphics_CGPoint { get; }
+	public Type CoreGraphics_CGRect { get; }
+	public Type CoreGraphics_CGSize { get; }
+
+	// optional if AddressBook present
+	public Type? ABAddressBook { get; }
+	public Type? ABPerson { get; }
+	public Type? ABRecord { get; }
+
+	// optional if AudioToolbox present
+	public Type? MusicSequence { get; }
+
+	// optional if AudioUnit present 
+	public Type? AudioComponent { get; }
+	public Type? AudioUnit { get; }
+	public Type? AURenderEventEnumerator { get; }
+
+	// optional if CoreAnimation present
+	public Type? CATransform3D { get; }
+
+	// optional if OpenGL present
+	public Type? CGLContext { get; }
+	public Type? CGLPixelFormat { get; }
+
+	// optional if CoreLocation present
+	public Type? CLLocationCoordinate2D { get; }
+
+	// optional if CoreMedia present
+	public Type? CMAudioFormatDescription { get; }
+	public Type? CMClock { get; }
+	public Type? CMFormatDescription { get; }
+	public Type? CMSampleBuffer { get; }
+	public Type? CMTime { get; }
+	public Type? CMTimebase { get; }
+	public Type? CMTimeMapping { get; }
+	public Type? CMTimeRange { get; }
+	public Type? CMVideoFormatDescription { get; }
+	public Type? CMVideoDimensions { get; }
+
+	// optional if CoreVideo present
+	public Type? CVImageBuffer { get; }
+	public Type? CVPixelBuffer { get; }
+	public Type? CVPixelBufferPool { get; }
+
+	// optional if CoreMidi present
+	public Type? MidiEndpoint { get; }
+
+	// optional if MapKit present
+	public Type? MKCoordinateSpan { get; }
+
+	// optional if MediaToolbox is present 
+	public Type? MTAudioProcessingTap { get; }
+
+	// optional if UIKit is present
+	public Type? UIOffset { get; }
+	public Type? UIEdgeInsets { get; }
+	public Type? NSDirectionalEdgeInsets { get; }
+
+	public TypeCache (BindingTouch bindingTouch, Assembly apiAssembly, Assembly corlibAssembly, Assembly platformAssembly)
+	{
+		if (bindingTouch.Frameworks is null)
+			throw ErrorHelper.CreateError (3, bindingTouch.CurrentPlatform);
+		if (bindingTouch.universe is null)
+			throw ErrorHelper.CreateError (4, bindingTouch.CurrentPlatform);
+
+		/* corlib */
+		System_Attribute = Lookup (corlibAssembly, "System", "Attribute");
```

Comment:
You may want to consider using `const string` for all the repeated strings below.

Reply:
That's a good point, I just brought this logic over wholesale. I'll get to it in an upcoming PR!